src/Annotation/Depend.php

```diff
@@ -2,26 +2,13 @@
 
 namespace Asynit\Annotation;
 
-use Doctrine\Common\Annotations\Annotation\Target;
-
-/**
- * @Annotation
- * @Target({"METHOD"})
- */
+#[\Attribute(\Attribute::TARGET_METHOD | \Attribute::IS_REPEATABLE)]
 class Depend
 {
-    private $dependency;
-
-    public function __construct(array $dependency)
-    {
-        $this->dependency = $dependency['value'];
-    }
+    public string $dependency;
 
-    /**
-     * @return mixed
-     */
-    public function getDependency()
+    public function __construct(string $dependency)
```

Comment:
php8, so declare dependency directly in constructo parameter ?

Reply:
When you do: 

```
#[Depend("a_test_method")]
```

yes it use the constructor, (and i believe it always use), but since php8 supported named parametere you can also do this

```
#[Depend(dependency: "a_test_method")]
```

Reply:
@bastnic mean:
```suggestion
    public function __construct(public string $dependency)
```